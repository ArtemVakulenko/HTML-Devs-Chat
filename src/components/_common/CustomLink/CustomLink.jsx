import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { useTranslation } from 'react-i18next';

const CustomLink = ({ path, title }) => {
  const { t } = useTranslation();
  return (
    <Link to={path}>
      {t(title)}
    </Link>
  );
};

export default CustomLink;

CustomLink.propTypes = {
  path: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};
