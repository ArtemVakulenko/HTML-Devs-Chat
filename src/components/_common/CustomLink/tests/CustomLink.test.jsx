import React from 'react';
import CustomLink from '../CustomLink';

jest.mock('react-i18next', () => ({
  useTranslation: jest.fn().mockImplementation(() => ({
    t: jest.fn(),
  })),
}),
);
jest.mock('react-router-dom', () => ({
  Link: jest.fn()
    .mockImplementation(
      (props) => <a href={props.path} >{props.children}</a>,
      ),
}));
describe('Title', () => {
  const props = {
    path: '/',
    title: 'This is link',
  };
  it('Should match snapshot', () => {
    const component = global.shallow(<CustomLink {...props} />);
    expect(component.html()).toMatchSnapshot();
  });
  it('Should have a', () => {
    const component = global.mount(<CustomLink {...props} />);
    expect(component.find('a')).toHaveLength(1);
  });
});
