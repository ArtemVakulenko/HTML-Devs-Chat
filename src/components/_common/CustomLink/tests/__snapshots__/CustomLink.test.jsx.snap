// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Should match snapshot 1`] = `"<a></a>"`;
